src/builders/rollup/plugins/cjs.ts

```diff
@@ -66,3 +66,298 @@ function CJSToESM(code: string): { code: string; map: any } | null {
     map: s.generateMap(),
   };
 }
+
+interface ParsedExports {
+  defaultExport: ESMExport;
+  defaultAlias: string;
+  exports: string[];
+}
+
+function extractExports(
+  code: string,
+  info: RenderedChunk,
+  ctx: BuildContext,
+): ParsedExports | undefined {
+  const defaultExport = findExports(code).find((e) =>
+    e.names.includes("default"),
+  );
+
+  if (!defaultExport) {
+    ctx.warnings.add(
+      `No default export found in ${info.fileName}, it contains default export but cannot be parsed.`,
+    );
+    return;
+  }
+
+  const regexp = /export\s*\{([^}]*)\}/;
+  const defaultExportRegexp = /\s*as\s+default\s*/;
+  const match = defaultExport.code.match(regexp);
+  if (!match?.length) {
+    ctx.warnings.add(
+      `No default export found in ${info.fileName}, it contains default export but cannot be parsed.`,
+    );
+    return;
+  }
+
+  let defaultAlias: string | undefined;
+  const exportsEntries: string[] = [];
+  for (const exp of match[1].split(",").map((e) => e.trim())) {
+    if (exp === "default") {
+      defaultAlias = exp;
+      continue;
+    }
+    const m = exp.match(defaultExportRegexp);
+    if (m) {
+      defaultAlias = exp.replace(m[0], "");
+    } else {
+      exportsEntries.push(exp);
+    }
+  }
+
+  if (!defaultAlias) {
+    ctx.warnings.add(
+      `No default export found in ${info.fileName}, it contains default export but cannot be parsed.`,
+    );
+    return;
+  }
+
+  return {
+    defaultExport,
+    defaultAlias,
+    exports: exportsEntries,
+  };
+}
+
+// export { default } from "magic-string";
+// or
+// import MagicString from 'magic-string';
+// export default MagicString;
+function handleDefaultCJSExportAsDefault(
+  code: string,
+  { defaultExport, exports }: ParsedExports,
+  imports: ParsedStaticImport[],
+  defaultImport?: ParsedStaticImport,
+): string | undefined {
+  if (defaultImport) {
+    return exports.length === 0
+      ? code.replace(
+          defaultExport.code,
+          `export = ${defaultImport.defaultImport}`,
+        )
+      : code.replace(
+          defaultExport.code,
+          `export = ${defaultImport.defaultImport};\nexport { ${exports.join(", ")} } from '${defaultExport.specifier}'`,
+        );
+  } else {
+    const magicString = new MagicString(code);
+    // add the import after last import in the code
+    const lastImportPosition =
+      imports.length > 0 ? imports.at(-1)?.end || 0 : 0;
+    if (lastImportPosition > 0) {
+      magicString.appendRight(
+        lastImportPosition,
+        `\nimport _default from '${defaultExport.specifier}';\n`,
+      );
+    } else {
+      magicString.prepend(
+        `import _default from '${defaultExport.specifier}';\n`,
```

Comment:
should we parse imports to avoid collisions or just use some suffix? if someone using `import { x as _default } from ...`

for example: `_default$unbuild`